services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/gdpr/GdprController.kt

```diff
@@ -77,18 +72,12 @@ class GdprController(
         }
 
         logger.info { "Finding GDPR information for user $userId" }
-        val applications = applicationService.getAllApplicationsForUser(userId)
-        val hankkeet = hankeService.loadHankkeetByUserId(userId)
-        if (hankkeet.isEmpty() && applications.isEmpty()) {
-            logger.warn { "No GDPR information found for user $userId" }
-            throw UserNotFoundException(userId)
-        }
+        val applications = applicationService.getAllApplicationsCreatedByUser(userId)
 
-        val userInfo = profiiliClient.getInfo(userId)
-        val gdprInfo = gdprJsonConverter.createGdprJson(applications, hankkeet, userInfo)
+        val gdprInfo = GdprJsonConverter.createGdprJson(applications, userId)
 
         if (gdprInfo == null) {
-            logger.warn { "No matching information found for GDPR request for user $userId" }
+            logger.warn { "No applications found for user $userId" }
```

Comment:
ResponseEntity is not the best option if errors return content different from normal responses. In Haitaton, this is true for most endpoints, since we mostly return HankeError error codes. Using something like `ResponseEntity<Any>` gives up most of the benefits we get from using a language with static typing. Also, many exceptions are thrown from the service layer, so we need exception handling anyway.

Exceptions shouldn't be used for _normal_ flow control. We want to exit the normal flow as soon we see an error, so exceptions are a valid way to handle them.

The cost of exceptions comes mostly from filling in the stack trace. This takes a couple of microseconds. I don't think that's very significant unless we throw and catch them in a loop or something. But this would be using them for normal control flow, which we can all agree on is bad.

That said, using a ResponseEntity is a valid approach for the 404 case since it doesn't return a body. I'll make the change to using it here.

Reply:
Changed as suggested.